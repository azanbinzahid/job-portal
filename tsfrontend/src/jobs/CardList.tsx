import React, { FC } from "react";
import styled from "styled-components";
import { Card, ListGroup } from "react-bootstrap";

const Title = styled.div`
  color: white;
`;

type Props = {
  title: string;
  list: string[];
};

const CardList: FC<Props> = ({ title, list }) => {
  console.log(list);
  return (
    <Card bg="primary">
      <Card.Header>
        <Title>{title}(s) </Title>
      </Card.Header>
      <ListGroup variant="flush">
        {list
          ? list.map((ele, index) => (
              <ListGroup.Item key={index}>{ele}</ListGroup.Item>
            ))
          : null}
      </ListGroup>
    </Card>
  );
};

export default CardList;
