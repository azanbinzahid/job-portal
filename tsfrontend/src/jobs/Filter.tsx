import React, { useState, useEffect, FC } from "react";
import Select from "react-select";
import { searchQuery } from "utils/helper";
import { withRouter, useLocation, RouteComponentProps } from "react-router-dom";

interface Props extends RouteComponentProps<any> {
  filterName: String;
  options: { value: String; label: String }[];
}

const Filter: FC<Props> = (props) => {
  const { filterName, options } = props;
  const [selectedOption, setSelectedOption] = useState([]);
  let params = useLocation().search.split("&");
  let query = searchQuery(params, filterName);

  useEffect(() => {
    let q = query;
    if (selectedOption) {
      selectedOption.forEach((element: { value: String; label: String }) => {
        q = q + `&${filterName}=${element.value}`;
      });
    }
    props.history.push(q);
  }, [selectedOption]);

  return (
    <div className="pb-5">
      <Select<any>
        isMulti
        label={filterName}
        defaultValue={selectedOption}
        onChange={setSelectedOption}
        options={options}
      />
    </div>
  );
};

export default withRouter(Filter);
