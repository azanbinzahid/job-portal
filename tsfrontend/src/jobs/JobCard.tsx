import React, { FC } from "react";
import { Button, Card } from "react-bootstrap";
import { Job } from "redux/types";
import { truncate } from "utils/helper";
import { withRouter, RouteComponentProps } from "react-router-dom";
import { FaMapMarkerAlt, FaStream } from "react-icons/fa";
import { HiOfficeBuilding } from "react-icons/hi";

interface Props extends RouteComponentProps<any> {
  job: Job;
}

const JobCard: FC<Props> = (props) => {
  const { job, history } = props;
  return (
    <Card border="primary">
      <Card.Header>
        <FaStream /> {job.category.join(", ")}
      </Card.Header>
      <Card.Body>
        <Card.Title className="font-weight-bold">{job.title}</Card.Title>
        <Card.Subtitle className="mb-2 text-muted">
          <HiOfficeBuilding /> {job.company}
        </Card.Subtitle>
        <Card.Text>{truncate(job.description)}</Card.Text>
      </Card.Body>
      <Card.Footer>
        <FaMapMarkerAlt /> {job.location.join(", ")}
        <Button
          style={{
            float: "right",
            padding: "revert",
          }}
          onClick={() => {
            history.push(`/jobs/${job.id}`);
          }}
          variant="outline-primary"
        >
          View Details
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default withRouter(JobCard);
