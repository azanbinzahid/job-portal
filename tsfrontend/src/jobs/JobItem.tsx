import React, { FC } from "react";
import { Button, Card, ListGroup } from "react-bootstrap";
import { Job } from "redux/types";
import { truncate } from "utils/helper";
import { withRouter, RouteComponentProps } from "react-router-dom";

interface Props extends RouteComponentProps<any> {
  job: Job;
}

const JobItem: FC<Props> = (props) => {
  const { job, history } = props;
  return (
    <Card border="primary">
      <Card.Header>{job.category.map((cat) => cat + ", ")}</Card.Header>
      <Card.Body>
        <Card.Title>{job.title}</Card.Title>
        <Card.Subtitle className="mb-2 text-muted">{job.company}</Card.Subtitle>
        <Card.Text>{truncate(job.description)}</Card.Text>
        <Button
          onClick={() => {
            history.push(`/jobs/${job.id}`);
          }}
          variant="primary"
        >
          View Details
        </Button>
      </Card.Body>
      <ListGroup variant="flush">
        {job.location.map((loc, index) => (
          <ListGroup.Item key={index}>{loc}</ListGroup.Item>
        ))}
      </ListGroup>
    </Card>
  );
};

export default withRouter(JobItem);
