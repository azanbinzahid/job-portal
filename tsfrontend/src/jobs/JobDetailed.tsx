import React, { useEffect, useCallback, FC } from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import {
  Container,
  Button,
  Card,
  ListGroup,
  CardColumns,
  Spinner,
} from "react-bootstrap";
import { fetchJob, applyJob } from "redux/actions";
import { RootState } from "redux/reducers";
import { Job } from "redux/types";
import CardList from "./CardList";
import SectionHeading from "app/common/SectionHeading";

const Title = styled.div`
  color: white;
`;

type Props = {
  job: Job;
  username: string;
  applyJob: (jobId: string, msg: string, type: string) => void;
  fetchJob: (jobId: string) => void;
  match: {
    params: {
      jobId: string;
    };
  };
};
export const JobDetailed: FC<Props> = (props) => {
  const {
    params: { jobId },
  } = props.match;
  const { fetchJob, job } = props;

  useEffect(() => {
    fetchJob(jobId);
  }, [fetchJob, jobId]);

  const handleClick = useCallback(
    (msg, type) => {
      props.applyJob(jobId, msg, type);
    },
    [props, jobId]
  );

  if (job == null) {
    return (
      <Container<React.ElementType> align="center" className="p-5">
        <Spinner animation="border" />
      </Container>
    );
  }

  return (
    <Container>
      <Container<React.ElementType> align="center" className="pb-5">
        <SectionHeading title="Job Details" />
        {!job.applicants.includes(props.username) ? (
          <Button
            size="lg"
            onClick={() => handleClick("Applied", "success")}
            variant="primary"
          >
            Apply Now
          </Button>
        ) : (
          <Button
            onClick={() => handleClick("Application Withdrawn", "warning")}
            variant="secondary"
          >
            Withdraw Application
          </Button>
        )}
      </Container>

      <CardColumns>
        <Card bg="primary">
          <Card.Header>
            <Title> Job Title </Title>
          </Card.Header>
          <ListGroup variant="flush">
            <ListGroup.Item>{job.title}</ListGroup.Item>
          </ListGroup>
        </Card>

        <Card bg="primary">
          <Card.Header>
            <Title>Company </Title>
          </Card.Header>
          <ListGroup variant="flush">
            <ListGroup.Item>{job.company}</ListGroup.Item>
          </ListGroup>
        </Card>

        <CardList title="Location" list={job.location} />

        <Card bg="primary">
          <Card.Header>
            <Title>Job Description </Title>
          </Card.Header>
          <Card.Body style={{ background: "white" }}>
            <Card.Text style={{ textAlign: "justify" }}>
              {job.description}
            </Card.Text>
          </Card.Body>
        </Card>

        <CardList title="Category" list={job.category} />
        <CardList title="Qualification" list={job.qualification} />

        <Card bg="primary">
          <Card.Header>
            <Title>Salary </Title>
          </Card.Header>
          <ListGroup variant="flush">
            <ListGroup.Item>{job.salaray}</ListGroup.Item>
          </ListGroup>
        </Card>

        <Card bg="primary">
          <Card.Header>
            <Title>Experiance </Title>
          </Card.Header>
          <ListGroup variant="flush">
            <ListGroup.Item>{job.experiance}</ListGroup.Item>
          </ListGroup>
        </Card>
      </CardColumns>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  job: state.jobReducer.job,
  username: state.userReducer.user.username,
});

const mapDispatchToProps = {
  fetchJob,
  applyJob,
};

export default connect(mapStateToProps, mapDispatchToProps)(JobDetailed as any);
