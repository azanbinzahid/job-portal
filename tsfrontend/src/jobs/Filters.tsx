import React, { FC } from "react";
import { connect } from "react-redux";
import { Row, Col, Container } from "react-bootstrap";
import { RootState } from "redux/reducers";
import Filter from "jobs/Filter";

type Props = {
  filters: any;
};

export const Filters: FC<Props> = ({ filters }) => {
  return (
    <Container<React.ElementType> className="pt-2 pb-2" align="center">
      <Row>
        {Object.keys(filters).map((key) => {
          let temp: any = [];
          filters[key].forEach((e: any) => {
            temp.push({ value: e.name, label: e.name });
          });
          return (
            <Col xs={6} sm={6} md={3}>
              <Filter
                filterName={key.toLowerCase()}
                options={temp}
                isMulti={false}
              />
            </Col>
          );
        })}
      </Row>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  filters: state.jobReducer.filters,
});

export default connect(mapStateToProps)(Filters as any);
