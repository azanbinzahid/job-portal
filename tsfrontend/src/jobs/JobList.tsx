import React, { FC, useEffect } from "react";
import { connect } from "react-redux";
import JobItem from "jobs/JobItem";
import PageHeader from "app/common/PageHeader";
import Search from "jobs/Search";
import { Container, CardColumns } from "react-bootstrap";
import { RootState } from "redux/reducers";
import { Job } from "redux/types";
import { useLocation } from "react-router-dom";
import { fetchJobs } from "redux/actions";
import Filter from "jobs/Filter";

type Props = {
  jobs: Job[];
  filters: any;
  fetchJobs: (params: String) => void;
};

export const JobList: FC<Props> = ({ jobs, fetchJobs, filters }) => {
  let q = useLocation().search;
  useEffect(() => {
    fetchJobs(q);
  }, [fetchJobs, q]);

  return (
    <Container>
      <PageHeader title="Job Listing Page" />
      <Search />

      {Object.keys(filters).map((key) => {
        let temp: any = [];
        filters[key].forEach((e: any) => {
          temp.push({ value: e.name, label: e.name });
        });
        return <Filter filterName={key.toLowerCase()} options={temp} />;
      })}

      <CardColumns>
        {jobs ? jobs.map((job) => <JobItem key={job.id} job={job} />) : null}
      </CardColumns>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  jobs: state.jobReducer.jobs,
  filters: state.jobReducer.filters,
});

const mapDispatchToProps = { fetchJobs };

export default connect(mapStateToProps, mapDispatchToProps)(JobList as any);
