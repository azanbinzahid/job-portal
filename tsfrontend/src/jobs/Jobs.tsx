import React, { FC } from "react";
import SectionHeading from "app/common/SectionHeading";
import Search from "jobs/Search";
import Filters from "jobs/Filters";
import JobList from "jobs/JobList";
import { RootState } from "redux/reducers";
import { connect } from "react-redux";

type Props = {
  jobCount: number;
};

export const Jobs: FC<Props> = ({ jobCount }) => {
  return (
    <>
      <SectionHeading
        title="Jobs Search"
        subtitle={jobCount.toString() + " results found"}
      />
      <Search />
      <Filters />
      <JobList />
    </>
  );
};
const mapStateToProps = (state: RootState) => ({
  jobCount: state.jobReducer.jobs.length,
});

export default connect(mapStateToProps)(Jobs as any);
