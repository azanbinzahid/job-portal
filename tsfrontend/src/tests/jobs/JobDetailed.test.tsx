import React from "react";
import { shallow } from "enzyme";
import { JobDetailed } from "jobs/JobDetailed";

const job = {
  id: 1,
  location: ["Lahore"],
  category: ["Customer Service"],
  qualification: ["Bachelors"],
  applicants: ["admin"],
  company: "Modern Tech",
  title: "WordPress Developer",
  salaray: 60000,
  description: "This is test job description",
  experiance: 1,
  datestamp: new Date(11 / 11 / 11),
};

test("renders the component", () => {
  const component = shallow(
    <JobDetailed
      job={job}
      username="user"
      applyJob={jest.fn()}
      fetchJob={jest.fn()}
      match={{
        params: {
          jobId: "",
        },
      }}
    />
  );
  expect(component).toMatchSnapshot();
});
