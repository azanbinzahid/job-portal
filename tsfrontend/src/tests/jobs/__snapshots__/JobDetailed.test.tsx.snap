// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Container
  fluid={false}
>
  <Jumbotron
    align="center"
    fluid={false}
  >
    <h1>
       Job Details Page 
    </h1>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Apply Now
    </Button>
  </Jumbotron>
  <CardColumns>
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
           Job Title 
        </styled.div>
      </CardHeader>
      <ListGroup
        variant="flush"
      >
        <ListGroupItem
          active={false}
          disabled={false}
        >
          WordPress Developer
        </ListGroupItem>
      </ListGroup>
    </Card>
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
          Company 
        </styled.div>
      </CardHeader>
      <ListGroup
        variant="flush"
      >
        <ListGroupItem
          active={false}
          disabled={false}
        >
          Modern Tech
        </ListGroupItem>
      </ListGroup>
    </Card>
    <CardList
      list={
        Array [
          "Lahore",
        ]
      }
      title="Location"
    />
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
          Job Description 
        </styled.div>
      </CardHeader>
      <CardBody
        style={
          Object {
            "background": "white",
          }
        }
      >
        <CardText
          style={
            Object {
              "textAlign": "justify",
            }
          }
        >
          This is test job description
        </CardText>
      </CardBody>
    </Card>
    <CardList
      list={
        Array [
          "Customer Service",
        ]
      }
      title="Category"
    />
    <CardList
      list={
        Array [
          "Bachelors",
        ]
      }
      title="Qualification"
    />
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
          Salary 
        </styled.div>
      </CardHeader>
      <ListGroup
        variant="flush"
      >
        <ListGroupItem
          active={false}
          disabled={false}
        >
          60000
        </ListGroupItem>
      </ListGroup>
    </Card>
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
          Experiance 
        </styled.div>
      </CardHeader>
      <ListGroup
        variant="flush"
      >
        <ListGroupItem
          active={false}
          disabled={false}
        >
          1
        </ListGroupItem>
      </ListGroup>
    </Card>
  </CardColumns>
</Container>
`;

exports[`renders the component 2`] = `
<Container
  fluid={false}
>
  <Jumbotron
    align="center"
    fluid={false}
  >
    <h1>
       Job Details Page 
    </h1>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="secondary"
    >
      Withdraw Application
    </Button>
  </Jumbotron>
  <CardColumns>
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
           Job Title 
        </styled.div>
      </CardHeader>
      <ListGroup
        variant="flush"
      >
        <ListGroupItem
          active={false}
          disabled={false}
        >
          WordPress Developer
        </ListGroupItem>
      </ListGroup>
    </Card>
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
          Company 
        </styled.div>
      </CardHeader>
      <ListGroup
        variant="flush"
      >
        <ListGroupItem
          active={false}
          disabled={false}
        >
          Modern Tech
        </ListGroupItem>
      </ListGroup>
    </Card>
    <CardList
      list={
        Array [
          "Lahore",
        ]
      }
      title="Location"
    />
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
          Job Description 
        </styled.div>
      </CardHeader>
      <CardBody
        style={
          Object {
            "background": "white",
          }
        }
      >
        <CardText
          style={
            Object {
              "textAlign": "justify",
            }
          }
        >
          This is test job description
        </CardText>
      </CardBody>
    </Card>
    <CardList
      list={
        Array [
          "Customer Service",
        ]
      }
      title="Category"
    />
    <CardList
      list={
        Array [
          "Bachelors",
        ]
      }
      title="Qualification"
    />
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
          Salary 
        </styled.div>
      </CardHeader>
      <ListGroup
        variant="flush"
      >
        <ListGroupItem
          active={false}
          disabled={false}
        >
          60000
        </ListGroupItem>
      </ListGroup>
    </Card>
    <Card
      bg="primary"
      body={false}
    >
      <CardHeader>
        <styled.div>
          Experiance 
        </styled.div>
      </CardHeader>
      <ListGroup
        variant="flush"
      >
        <ListGroupItem
          active={false}
          disabled={false}
        >
          1
        </ListGroupItem>
      </ListGroup>
    </Card>
  </CardColumns>
</Container>
`;
